modules/asset-manager-workflowoperation/src/main/java/org/opencastproject/workflow/handler/assetmanager/AssetManagerDeleteWorkflowOperationHandler.java

```diff
@@ -73,9 +78,22 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
     final MediaPackage mediaPackage = workflowInstance.getMediaPackage();
     final String mpId = mediaPackage.getIdentifier().toString();
 
+    WorkflowOperationInstance currentOperation = workflowInstance.getCurrentOperation();
+    boolean keepLastSnapshot = BooleanUtils.toBoolean(currentOperation.getConfiguration(OPT_LAST_SNAPSHOT));
+
     try {
       final AQueryBuilder q = assetManager.createQuery();
-      final long deleted = q.delete(DEFAULT_OWNER, q.snapshot()).where(q.mediaPackageId(mpId)).run();
+      final long deleted;
+
+      if (keepLastSnapshot) {
+        deleted = q.delete(DEFAULT_OWNER, q.snapshot())
+                .where(q.mediaPackageId(mpId).and(q.version().isLatest().not())).run();
+        logger.info("Deleting all but latest Snapshot %s", mpId);
```

Comment:
This renders as `Deleting all but latest Snapshot %s`

Reply:
@gregorydlogan Why. I do not see the difference to the old code 2 lines below. And that works.
If there is a right way i will correct this in this class. Can you give me a hint?

Reply:
Try:

        logger.info("Deleting all but latest Snapshot {}", mpId);

The code below is explicitly using format(...).

Reply:
My apologies. I had set the wrong upstream on my local git repo, corrected this here already. The magic of git, is not working with lack of coffee. 
Thanks @smarquard.

Reply:
changed logging output variables to {}.